File: claude_long/go/session_manager.go
package main

import (
	"fmt"
	"os"
	"os/exec"
	"path/filepath"
	"time"
)

// SessionManager 管理会话
type SessionManager struct {
	ProjectDir      string
	ProgressManager *ProgressManager
	FeatureManager  *FeatureManager
}

// NewSessionManager 创建新的会话管理器
func NewSessionManager(projectDir string) *SessionManager {
	if projectDir == "" {
		projectDir = "."
	}
	absPath, _ := filepath.Abs(projectDir)

	return &SessionManager{
		ProjectDir:      absPath,
		ProgressManager: NewProgressManager(filepath.Join(absPath, "claude-progress.txt")),
		FeatureManager:  NewFeatureManager(filepath.Join(absPath, "feature_list.json")),
	}
}

// InitializeProject 初始化新项目
func (sm *SessionManager) InitializeProject() error {
	fmt.Println("🚀 Initializing long-running agent project...")

	// 初始化 Git
	gitDir := filepath.Join(sm.ProjectDir, ".git")
	if _, err := os.Stat(gitDir); os.IsNotExist(err) {
		fmt.Println("Initializing Git repository...")
		cmd := exec.Command("git", "init")
		cmd.Dir = sm.ProjectDir
		if err := cmd.Run(); err != nil {
			fmt.Printf("Warning: Failed to initialize git: %v\n", err)
		}
	}

	// 初始化进度文件
	fmt.Println("Creating progress file...")
	if err := sm.ProgressManager.Initialize(); err != nil {
		return err
	}

	// 初始化功能列表
	fmt.Println("Creating feature list...")
	if err := sm.FeatureManager.Initialize(nil); err != nil {
		return err
	}

	// 添加初始会话记录
	sm.ProgressManager.AddSession(Session{
		Date: time.Now(),
		CompletedWork: []string{
			"Initialized project structure",
			"Created progress tracking file",
			"Set up feature list",
		},
		Status: "Project initialized and ready for development",
		NextSteps: []string{
			"Begin working on first feature",
		},
	})

	fmt.Println("\n✅ Project initialized successfully!")
	fmt.Printf("📁 Project directory: %s\n", sm.ProjectDir)
	return nil
}

// StartSession 开始新的编码会话
func (sm *SessionManager) StartSession() error {
	fmt.Println("🔄 Starting new coding session...")
	fmt.Printf("📁 Working directory: %s\n\n", sm.ProjectDir)

	// 读取进度
	fmt.Println("📖 Reading progress file...")
	progress, err := sm.ProgressManager.GetRecentSessions(3)
	if err != nil {
		return err
	}
	for _, session := range progress {
		if len(session) > 200 {
			fmt.Println(session[:200] + "...")
		} else {
			fmt.Println(session)
		}
	}

	// 读取功能列表
	fmt.Println("\n📋 Loading feature list...")
	summary, err := sm.FeatureManager.GetSummary()
	if err != nil {
		return err
	}
	fmt.Printf("Total features: %v\n", summary["total"])
	fmt.Printf("Completed: %v\n", summary["completed"])
	fmt.Printf("Pending: %v\n", summary["pending"])

	// 获取下一个功能
	fmt.Println("\n🎯 Next feature to work on:")
	nextFeature, err := sm.FeatureManager.GetNextPendingFeature()
	if err != nil {
		return err
	}
	if nextFeature != nil {
		fmt.Printf("ID: %s\n", nextFeature.ID)
		fmt.Printf("Priority: %d\n", nextFeature.Priority)
		fmt.Printf("Description: %s\n", nextFeature.Description)
		fmt.Println("\nImplementation steps:")
		for i, step := range nextFeature.Steps {
			fmt.Printf("  %d. %s\n", i+1, step)
		}
	} else {
		fmt.Println("🎉 All features completed!")
	}

	fmt.Println("\n✅ Session started! Remember to:")
	fmt.Println("  1. Work on ONE feature at a time")
	fmt.Println("  2. Test thoroughly before marking complete")
	fmt.Println("  3. Commit changes with descriptive messages")

	return nil
}

// EndSession 结束当前会话
func (sm *SessionManager) EndSession(summary string) error {
	fmt.Println("🏁 Ending session...")

	err := sm.ProgressManager.AddSession(Session{
		Date:          time.Now(),
		CompletedWork: []string{summary},
		Status:        "Session completed",
	})
	if err != nil {
		return err
	}

	fmt.Println("\n✅ Session ended successfully!")
	return nil
}

File: claude_long/python/session_manager.py
"""
Session Manager - 会话管理器 (Python 实现)

整合进度管理、功能管理，提供完整的会话生命周期管理。
"""

import os
import subprocess
from datetime import datetime
from typing import Dict, List

from progress_manager import ProgressManager
from feature_manager import FeatureManager


class SessionManager:
    """管理长期运行代理会话的类"""

    def __init__(self, project_dir: str = "."):
        self.project_dir = os.path.abspath(project_dir)
        self.progress_manager = ProgressManager(os.path.join(project_dir, "claude-progress.txt"))
        self.feature_manager = FeatureManager(os.path.join(project_dir, "feature_list.json"))

    def initialize_project(self) -> None:
        """初始化新项目"""
        print("🚀 Initializing long-running agent project...")

        # 初始化 Git
        if not os.path.exists(os.path.join(self.project_dir, ".git")):
            self._run_command("git init")

        # 初始化文件
        self.progress_manager.initialize()
        self.feature_manager.initialize()

        print("✅ Project initialized successfully!")

    def start_session(self) -> Dict:
        """开始新的编码会话"""
        print("🔄 Starting new coding session...")
        print(f"📁 Working directory: {self.project_dir}\n")

        # 读取进度
        print("📖 Reading progress...")
        progress = self.progress_manager.get_recent_sessions(3)
        for session in progress:
            print(session[:200] + "..." if len(session) > 200 else session)

        # 读取功能
        print("\n📋 Loading features...")
        summary = self.feature_manager.get_summary()
        print(f"Total: {summary['total']}, Completed: {summary['completed']}, Pending: {summary['pending']}")

        # 下一个功能
        print("\n🎯 Next feature:")
        next_feature = self.feature_manager.get_next_pending_feature()
        if next_feature:
            print(f"[{next_feature['id']}] {next_feature['description']}")
        else:
            print("All features completed!")

        return {'progress': progress, 'summary': summary, 'next_feature': next_feature}

    def end_session(self, summary: str) -> None:
        """结束当前会话"""
        print("🏁 Ending session...")
        self.progress_manager.add_session(completed_work=[summary], status="Session completed")
        print("✅ Session ended successfully!")

    def _run_command(self, command: str) -> str:
        """运行命令"""
        try:
            result = subprocess.run(command, shell=True, cwd=self.project_dir,
                                  capture_output=True, text=True)
            return result.stdout
        except Exception as e:
            print(f"Error: {e}")
            return ""


if __name__ == "__main__":
    import sys
    if len(sys.argv) < 2:
        print("Usage: python session_manager.py <command>")
        print("Commands: init, start, end")
        sys.exit(1)

    sm = SessionManager()
    command = sys.argv[1]

    if command == "init":
        sm.initialize_project()
    elif command == "start":
        sm.start_session()
    elif command == "end":
        summary = ' '.join(sys.argv[2:]) if len(sys.argv) > 2 else "Session completed"
        sm.end_session(summary)
